src/client/executor.rs

```diff
@@ -147,6 +164,14 @@ impl Client {
             }
         };
 
+        // If the transaction is being aborted, unpin session before retry and regardless of whether
+        // the command succeeds, fails, or is executed.
+        if let Some(ref mut session) = session {
```

Comment:
Hm, that's a good point. The unpin has to happen after server selection, but it should be agnostic of whether the operation is executed, if I'm understanding the behavior correctly. How do you feel about leaving an unpin here (so server selection can occur as expected) and another at the end of `abort_transaction`?

I'm thinking we should write an internal test to verify the behavior in this case, since it's probably the strangest unpin.

Reply:
Looks good! I think we can remove this unpin here now.